nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestCommentManager.java

```diff
@@ -89,8 +119,15 @@
     @Inject
     protected CoreFeature coreFeature;
 
+    @Inject
+    protected NotificationService notificationService;
+
     public abstract Class<? extends CommentManager> getType();
 
+    protected DocumentRef getCommentedDocRef(CoreSession session, DocumentModel commentDocModel, boolean reply) {
+        throw new UnsupportedOperationException();
```

Comment:
Why not abstract?

Reply:
its to avoid any backward compatibility in a case where a customer inherent from this abstract class.

Reply:
I'm not sure we want to follow this on tests 🤔. It could deserve a discussion with the team.

In the meanwhile, you already broke the backward compatibility with:
```
public abstract Class<? extends CommentManager> getType();
```
We can probably do it a second time 😉

Reply:
> public abstract Class<? extends CommentManager> getType();
i agree, sorry i thought that this the abstract class and not the test. and in a test i agree no need to keep backward compatibility ;)